### PR TITLE
added end-to-end tileset test

### PR DESCRIPTION
**Pull request by mattelser**
Created an end-to-end test for tileset loading, including the first c++ side event subscription in the codebase.

### CODE REVIEW

File: tests/src/tilesetTests.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+#include <carb/dictionary/DictionaryUtils.h>
+#include <carb/events/IEvents.h>
+#include <doctest/doctest.h>
+#include <omni/kit/IApp.h>
+
+pxr::SdfPath endToEndTilesetPath;
+bool endToEndTilesetLoaded = false;
+carb::events::ISubscriptionPtr endToEndTilesetSubscriptionPtr;
+class TilesetLoadListener;
+TilesetLoadListener* tilesetLoadListener;
```
Comment (mattelser) [inline tests/src/tilesetTests.cpp:18]: It felt weird not to use a smart pointer, but for a global I didn't think they made sense

Comment (lilleyse) [reply tests/src/tilesetTests.cpp:18]: I would still switch to a `unique_ptr` just for internal consistency. In the cleanup function you can call `reset` on it to free the memory.

**Review by lilleyse — COMMENTED** [review]
Looks good, just one comment

File: tests/src/tilesetTests.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+using namespace cesium::omniverse;
+
+class TilesetLoadListener final : public carb::events::IEventListener {
+  public:
+    size_t refCount = 0;
```
[the remark this answers is not in the document]
Comment (mattelser) [reply tests/src/tilesetTests.cpp:26]: It's mostly there so the virtual `addRef()` and `release()` have something to return

**Review by lilleyse — APPROVED** [review]

**Pull request merged**